platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/MarkerViewManager.java

```diff
@@ -503,6 +503,21 @@ public void invalidateViewMarkersInVisibleRegion() {
                 }
               }
 
+              adaptedView.setOnClickListener(new View.OnClickListener() {
```

Comment:
yes, this has been on my list todo for a while, this method should be split up into smaller methods 👍